### ISSUE TITLE
The setcontent function for twig allows limit but not offset

### ISSUE THREAD

**Issue by ToBe998** [bolt/bolt#5254]
The current Twig fiunction `setcontent` allows the parameter `limit` to limit the number of returned content. Sadly it does not support a `offset` parameter. So it is currently not possible to implement some kind of _paging_ for your content (other than the built in pagination).

This is especially problematic as the built in paging function is not sufficient for websites bigger than small blogs:
- The pager GET parameter does not follow googles best practice of `page=1` making it unusable for seo as it can't detect paginated content on it's own.
- The links/buttons do not set rel=prev/next attributes leading to duplicate content and/or seo canibalization

In short, the built in pagination is ok for small websites that dont need seo, but unusable for bussiness sites wich need seo and enhanced tracking. It's perfectly acceptable that such sites have to create their own pager, but to do so, they need a _offset_ parameter.

Suggestion:

```
{% setcontent mypages = 'pages' limit 10 offset 20 orderby 'title' %}
```

Please dont use something like page since we need full control over the amount of items to skip with offset.

**Comment by bobdenotter** [comment]
Because of "Spring cleaning", we're going to close this issue. Unless somebody steps up to help out with implementing it, it's not expected that the core team will have the opportunity to work on this soon. Keeping it open might set false expectations for people stumbling upon this issue.

This is not because we don't see this feature as potentially worthwhile, it is more to do with an overload of potential new features that could be developed combined with non-limitless amounts of available time and resources. 

If you disagree, feel free to re-open for discussion.

**Comment by anketwokings** [comment]
I stumble onto the exact same issue where it looks that I need to build a custom pagination, and googling brought me here. Will look into it if there's a way around it.

**Comment by Spomsoree** [comment]
I could use this too. 
Is there any workaround, expect getting rid of them in a twig loop?

In my opinion this is a crucial feature, isn't it?

**Comment by ToBe998** [comment]
We are mostly using our own select content function these days. Would be interesting to know if the move to symfony 4 and all the refactoring will bring changes here?
